packages/@aws-cdk/core/lib/cfn-parse.ts

```diff
@@ -532,24 +582,34 @@ export class CfnParser {
       return leftHalf + CfnReference.for(refResource, attribute, true).toString() + this.parseFnSubString(rightHalf, map);
     }
   }
-}
 
-function specialCaseRefs(value: any): any {
-  switch (value) {
-    case 'AWS::AccountId': return Aws.ACCOUNT_ID;
-    case 'AWS::Region': return Aws.REGION;
-    case 'AWS::Partition': return Aws.PARTITION;
-    case 'AWS::URLSuffix': return Aws.URL_SUFFIX;
-    case 'AWS::NotificationARNs': return Aws.NOTIFICATION_ARNS;
-    case 'AWS::StackId': return Aws.STACK_ID;
-    case 'AWS::StackName': return Aws.STACK_NAME;
-    case 'AWS::NoValue': return Aws.NO_VALUE;
-    default: return undefined;
+  private specialCaseRefs(value: any): any {
+    if (value in this.parameters) {
+      return this.parameters[value];
+    }
+    switch (value) {
+      case 'AWS::AccountId': return Aws.ACCOUNT_ID;
+      case 'AWS::Region': return Aws.REGION;
+      case 'AWS::Partition': return Aws.PARTITION;
+      case 'AWS::URLSuffix': return Aws.URL_SUFFIX;
+      case 'AWS::NotificationARNs': return Aws.NOTIFICATION_ARNS;
+      case 'AWS::StackId': return Aws.STACK_ID;
+      case 'AWS::StackName': return Aws.STACK_NAME;
+      case 'AWS::NoValue': return Aws.NO_VALUE;
+      default: return undefined;
+    }
   }
-}
 
-function specialCaseSubRefs(value: string): string | undefined {
-  return value.indexOf('::') === -1 ? undefined: '${' + value + '}';
+  private specialCaseSubRefs(value: string): string | undefined {
+    if (value in this.parameters) {
+      return this.parameters[value];
+    }
+    return value.indexOf('::') === -1 ? undefined: '${' + value + '}';
+  }
+
+  private get parameters(): { [parameterName: string]: any } {
+    return this.options.parameters || {};
+  }
 }
```

Comment:
Can you believe how much simpler the changes in this file are now, vs. the initially proposed version??

Reply:
yep this is substantially cleaner.